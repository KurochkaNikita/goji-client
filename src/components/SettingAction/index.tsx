import { MouseEvent } from "react";
import DeleteIcon from "@mui/icons-material/MoreVert";
import IconButton from "@mui/material/IconButton";

interface ISettingAction {
  onClick: (e: MouseEvent<HTMLButtonElement>) => void;
}

function SettingAction(props: ISettingAction) {
  const { onClick } = props;

  const handlerClick = (e: MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    e.stopPropagation();
    onClick(e);
  };

  return (
    <IconButton onClick={handlerClick} edge="end" aria-label="delete">
      <DeleteIcon />
    </IconButton>
  );
}

export default SettingAction;
