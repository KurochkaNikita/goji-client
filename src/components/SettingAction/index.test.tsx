import { fireEvent, render, screen } from "@testing-library/react";

import SettingAction from "./index";

describe("Header", () => {
  const onClickMock = jest.fn();

  it("Snapshot", () => {
    const { container } = render(<SettingAction onClick={onClickMock} />);
    expect(container).toMatchSnapshot();
  });

  it("should render the icon button and handle click event", () => {
    render(<SettingAction onClick={onClickMock} />);
    const button = screen.getByRole("button", { name: /delete/i });
    expect(button).toBeInTheDocument();
    fireEvent.click(button);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
