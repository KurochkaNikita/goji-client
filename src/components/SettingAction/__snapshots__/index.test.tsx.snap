// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot 1`] = `
<div>
  <button
    aria-label="delete"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-oeq6yz-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
      data-testid="MoreVertIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
      />
    </svg>
  </button>
</div>
`;
